src/Mvc/test/Mvc.IntegrationTests/ComplexTypeIntegrationTestBase.cs

```diff
@@ -0,0 +1,3774 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Binders;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.IntegrationTests
+{
+    // Integration tests targeting the behavior of the ComplexTypeModelBinder and related classes
+    // with other model binders.
+    public abstract class ComplexTypeIntegrationTestBase
+    {
+        private const string AddressBodyContent = "{ \"street\" : \"" + AddressStreetContent + "\" }";
+        private const string AddressStreetContent = "1 Microsoft Way";
+
+        private static readonly byte[] ByteArrayContent = Encoding.BigEndianUnicode.GetBytes("abcd");
+        private static readonly string ByteArrayEncoded = Convert.ToBase64String(ByteArrayContent);
+
+        protected abstract Type ExpectedModelBinderType { get; }
+
+        [Fact]
+        public void ExpectedModelBinderIsConstructed()
+        {
+            // Arrange
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Order1)
+            };
+            var testContext = GetTestContext();
+
+            var metadata = GetMetadata(testContext, parameter);
+
+            // Act
+            var modelBinder = GetModelBinder(testContext, parameter, metadata);
+
+            // Assert
+            Assert.Equal(ExpectedModelBinderType, modelBinder.GetType());
+        }
+
+        private class Order1
+        {
+            public int ProductId { get; set; }
+
+            public Person1 Customer { get; set; }
+        }
+
+        private class Person1
+        {
+            public string Name { get; set; }
+
+            [FromBody]
+            public Address1 Address { get; set; }
+        }
+
+        private class Address1
+        {
+            public string Street { get; set; }
+        }
+
+        [Fact]
+        public async Task MutableObjectModelBinder_BindsNestedPOCO_WithBodyModelBinder_WithPrefix_Success()
```

Comment:
nit: Could you please clean up the `MutableObjectModelBinder_` prefixes in the test names and the mention of that very old binder name in the comments❔